flink-table/flink-table-common/src/main/java/org/apache/flink/table/factories/CatalogStoreFactory.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.factories;
+
+import org.apache.flink.annotation.PublicEvolving;
+import org.apache.flink.configuration.ReadableConfig;
+import org.apache.flink.table.catalog.CatalogStore;
+
+import java.util.Map;
+
+/**
+ * A factory to create configured catalog store instances based on string-based properties. See also
+ * {@link Factory} for more information.
+ */
+@PublicEvolving
+public interface CatalogStoreFactory extends Factory {
+
+    /** Creates a {@link CatalogStore} instance from context information. */
+    CatalogStore createCatalogStore(Context context);
+
+    /** Initialize the CatalogStoreFactory. */
+    void open(Context context);
+
+    /** Close the CatalogStoreFactory. */
+    void close();
```

Comment:
Why a factory need the Initialization and tear-down methods? Please give more explanation to developers. Maybe we known the reason as the FLIP participates but other developer would confuse a lot.